### PR TITLE
Issue-1694: Boot screen should actually countdown the remaining seconds until reconnect

### LINKED ISSUE #1694: Boot screen should actually countdown the remaining seconds until reconnect
# Task Description

The `strongbox/strongbox-web-ui` can run independently from `strongbox/strongbox` during the development. When you start `strongbox-web-ui` via `npm start` or `ng serve` it becomes accessible at `localhost:4200`. When you hit that address you will see the "booting" screen which will try to connect to the backend at `localhost:48080` so it can report on the booting progress. 

Right now, if the UI cannot connect to the backend for some reason (i.e. it's down or is not ready to accept incoming requests) it will try to reconnect infinitely and will report a message saying how many seconds it will wait before retying.  

image

We need to make a timer to update the message on every second so that it would report `retrying in X-1` until it hits 0 or something interrupts the timer (like `this.destroy$`). This can be achieved using `rxjs`. Make sure to avoid memory leaks (i.e. use kill switch like `takeUntil`).

The main code for the booting screen is located in the [`boot-progress.service.ts`](https://github.com/strongbox/strongbox-web-ui/blob/c37138fd888ede6affaf32fc239a2ea7ed2808c1/src/app/modules/core/services/boot-progress.service.ts)

# Useful Links

* [rxjs](https://www.learnrxjs.io/learn-rxjs/operators/creation/interval)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @steve-todorov

### PR DESCRIPTION
**Pull request by rljoia**
Fixes strongbox/strongbox#1694

### CODE REVIEW

**Comment by steve-todorov** [comment]
@rljoia thank you very much for this PR! There are a few things I would like to mention:

1. Please fix your git author name and email so that they are recognized in github. This is important for the [ICLA](https://strongbox.github.io/contributing.html#submitting-pull-requests) as well since you need to commit under the email you signed the ICLA with.
2. In your PR you have made some unnecessary changes to the `retryWithConsecutiveBreak.ts` operator. All of the information you need is already available in the `attempt` variable in `onRetry` so you don't actually need to change anything.
https://github.com/strongbox/strongbox-web-ui/blob/cdc3569cd895a3772da68bf4ba20d5915f4da8dc/src/app/modules/core/services/boot-progress.service.ts#L95-L106
3. You are on the right track by using the `onRetry` in `BootProgressService` to change the message. However your code can be further improved and simplified for easier future maintenance:
   - Make a new method called `propagateRetryMessage` which accepts a `RetryAttempt` argument
   - Call this `propagateRetryMessage(attempt)` from `onRetry` (and move existing code into the new method)
   - You don't need to have both `interval` and `timer`. You can simply use `timer(0,1000)` which will immediately emit the first value and then emit every 1000ms (this is actually better than my proposal to use `interval` in the task description). You can then use the `take` pipe and the `attempt.delay` to kill the timer at the right moment. :)
   - Your changes seem to have broken the exponential delay - it is always starting at `15s`. It should actually exponentially increase the delay from `1s` to `15s` after each retry. This will fix itself when you revert `retryWithConsecutiveBreak` to what it was.
   - It would be nice for the user if he sees a `Connecting..` message after the timer hits `0` which can be achieved using `finalize` pipe in the `timer`.
 
Ping me in the chat and I'll give you some more hints about how to do all of this. :)

**Comment by carlspring** [comment]
@rljoia ,

Thank you for your contribution!

The username under which you're committing and pushing to Github seems to be mis-configured. You will have to look into how to configure your git settings properly, amend the author of the commits and then force push back to the branch. (When your user is properly set up, you see your avatar when against your commits in the commit history, otherwise it's a default grey avatar).

File: package.json [hunk 1]
```diff
@@ -60,10 +60,11 @@
     "zone.js": "~0.9.1"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "0.803.25",
-    "@angular/cli": "~8.3.25",
+    "@angular-devkit/build-angular": "0.803.26",
+    "@angular/cli": "~8.3.26",
     "@angular/compiler-cli": "8.1.0",
     "@angular/language-service": "8.1.0",
+    "@babel/compat-data": "~7.8.0",
```
Comment (steve-todorov) [inline package.json]: You don't need this fix - it's already included in the updated `@angular/cli` and `@angular-devkit/build-angular` :)

**Review by steve-todorov — APPROVED** [review]
LGTM! Thank you very much for fixing this! Oh... and congratulations on your first PR! :)

**Pull request merged**